### PR TITLE
Make signal handling simpler and work on linux

### PR DESCRIPTION
**Pull request by jechter**
Use sigaction rather than sigwait, which is simpler and works on linux

### CODE REVIEW

File: src/SignalHandler.cpp [hunk 1]
```diff
@@ -118,12 +104,15 @@ void SignalHandlerInit()
...
-        pthread_detach(sigthread);
-    }
+
+    struct sigaction s;
+
+    memset(&s, 0, sizeof(struct sigaction));
+    s.sa_handler = HandleSignal;
+    sigaction(SIGINT, &s, NULL);
+    sigaction(SIGTERM, &s, NULL);
+    sigaction(SIGQUIT, &s, NULL);
```
Comment (mostynb) [inline src/SignalHandler.cpp]: Should we check the return values?

File: src/BuildQueue.cpp [hunk 2]
```diff
@@ -97,8 +97,6 @@ void BuildQueueInit(BuildQueue *queue, const BuildQueueConfig *config, const cha
 
     Log(kDebug, "build queue initialized; ring buffer capacity = %u", queue->m_QueueCapacity);
 
-    // Block all signals on the main thread.
```
Comment (mostynb) [inline src/BuildQueue.cpp:100]: Do we know why the signals were being blocked here previously?

Comment (jechter) [reply src/BuildQueue.cpp:100]: Yes. The old signal handling code relied on:
1. Calling sigwait on a signal handling thread
2. Blocking signals on the main thread, so it does not get killed when a signal comes in.

We don't know why this approach was chosen over a more simple signal handler callback approach (With a signal handler, we don't need to block signals). But we did find that it did not work on linux, and this does.

File: src/SignalHandler.cpp (hunk 1 above)
Comment (jechter) [reply src/SignalHandler.cpp]: I guess that makes sense.

**Review by mostynb — APPROVED** [review]

**Pull request merged**